main.go

```diff
@@ -212,3 +241,86 @@ func checkLxcfsCfg() error {
 	}
 	return nil
 }
+
+// load daemon config file
+func loadDaemonFile(cfg *config.Config, flagSet *pflag.FlagSet) error {
+	configFile := cfg.ConfigFile
+	if configFile == "" {
+		return nil
+	}
+
+	contents, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to read contents from config file %s: %s", configFile, err)
+	}
+
+	var fileFlags map[string]interface{}
+	if err = json.NewDecoder(bytes.NewReader(contents)).Decode(&fileFlags); err != nil {
+		return fmt.Errorf("failed to decode json: %s", err)
+	}
+
+	if len(fileFlags) == 0 {
+		return nil
+	}
+
+	// check if invalid or unknow flag exist in config file
+	if err = getUnknowFlags(flagSet, fileFlags); err != nil {
+		return err
+	}
+
+	// check conflict in command line flags and config file
+	if err = getConflictConfigurations(flagSet, fileFlags); err != nil {
+		return err
+	}
+
+	fileConfig := &config.Config{}
+	if err = json.NewDecoder(bytes.NewReader(contents)).Decode(fileConfig); err != nil {
+		return fmt.Errorf("failed to decode json: %s", err)
+	}
+
+	// merge configurations from command line flags and config file
+	err = mergeConfigurations(fileConfig, cfg)
+	return err
+}
+
+// find unknow flag in config file
+func getUnknowFlags(flagSet *pflag.FlagSet, fileFlags map[string]interface{}) error {
+	var unknowFlags []string
+
+	for k := range fileFlags {
+		f := flagSet.Lookup(k)
+		if f == nil {
+			unknowFlags = append(unknowFlags, k)
+		}
+	}
+
+	if len(unknowFlags) > 0 {
+		return fmt.Errorf("unknow flags: %s", strings.Join(unknowFlags, ", "))
+	}
+
+	return nil
+}
+
+// find conflict in command line flags and config file
+func getConflictConfigurations(flagSet *pflag.FlagSet, fileFlags map[string]interface{}) error {
+	var conflictFlags []string
+	flagSet.Visit(func(f *pflag.Flag) {
+		if v, exist := fileFlags[f.Name]; exist {
+			conflictFlags = append(conflictFlags, fmt.Sprintf("conflict flag %s, from flag: %s, from config file: %s", f.Name, f.Value.String(), v.(string)))
+		}
+	})
+
+	if len(conflictFlags) > 0 {
+		return fmt.Errorf("get %s", strings.Join(conflictFlags, ", "))
```

Comment:
I think this error msg will be not so readable: `get confict flag a, from flag b, from config file: /etc/pouch/config.json, confict flag c, from flag d, from config file: /etc/pouch/config.json, confict flag e, from flag f, from config file: /etc/pouch/config.json`.

Could you make it more readable? @Ace-Tang

Reply:
maybe you can give a advise, @allencloud , I have no thought

Reply:
Maybe we can try to use:
```
conflictFlags = append(conflictFlags, fmt.Sprintf("conflict config option (%s) both from flag (%s) and config file %s", f.Name, f.Value.String(), v.(string)))
```

Then the error will be like:
```
get conflict config option (a) both from flag (b) and config file /etc/pouch/config.json, conflict config option (c) both from flag (d) and config file /etc/pouch/config.json
```

Reply:
how about  `found conflict flags in command line and config file, flag a, from flag b, from config file: /etc/pouch/config.json, confict flag c, from flag d, from config file: /etc/pouch/config.json`

Reply:
Since we can only set one config file, maybe we can simplify with this: `found conflict flags in command line and config file /etc/pouch/config.json: flag a and config b, flag c and config d...` @Ace-Tang

Reply:
good.